lib/Dialect/ModArith/IR/ModArithOps.td

```diff
@@ -68,6 +64,26 @@ def ModArith_ExtractOp : ModArith_Op<"extract", [Pure, ElementwiseMappable]> {
   let assemblyFormat = "operands attr-dict `:` type($input) `->` type($output)";
 }
 
+def ModArith_ModSwitchOp : ModArith_Op<"mod_switch", [Pure, SameOperandsAndResultShape]> {
+  let summary = "change the modulus of a mod_arith";
+
+  let description = [{
+    "mod_switch" operation to change the modulus of a mod_arith type to a
+    bigger space. Operation will not allow to switch to a smaller space.
```

Comment:
> Operation will not allow to switch to a smaller space.

I think we should be careful here in that modulus switching is rarely a ring homomorphism, meaning we cannot freely factor operations through this op (which would block possible rewrite patterns). That said, I see no problem having modulus switching to a smaller space.

You mention information loss, but I don't see this as a problem, since you might just be intended to project values onto a smaller ring (which is a valid thing to do).

Reply:
Check, I'll remove the verifier